Fail to compile with GCC11

### Observed behaviour

```
[ 18%] Building CXX object CMakeFiles/pioneer-lib.dir/src/Pi.cpp.o
In file included from /home/tle/src/pioneer/src/Json.h:7,
                 from /home/tle/src/pioneer/src/JsonUtils.h:9,
                 from /home/tle/src/pioneer/src/ship/Propulsion.h:8,
                 from /home/tle/src/pioneer/src/ShipType.h:7,
                 from /home/tle/src/pioneer/src/Intro.h:10,
                 from /home/tle/src/pioneer/src/Pi.cpp:20:
/home/tle/src/pioneer/contrib/json/json.hpp: In instantiation of ‘std::string nlohmann::detail::lexer<BasicJsonType>::get_token_string() const [with BasicJsonType = nlohmann::basic_json<>; std::string = std::__cxx11::basic_string<char>]’:
/home/tle/src/pioneer/contrib/json/json.hpp:4564:57:   required from ‘void nlohmann::detail::parser<BasicJsonType>::parse(bool, BasicJsonType&) [with BasicJsonType = nlohmann::basic_json<>]’
/home/tle/src/pioneer/contrib/json/json.hpp:17252:79:   required from ‘static nlohmann::basic_json<ObjectType, ArrayType, StringType, BooleanType, NumberIntegerType, NumberUnsignedType, NumberFloatType, AllocatorType, JSONSerializer> nlohmann::basic_json<ObjectType, ArrayType, StringType, BooleanType, NumberIntegerType, NumberUnsignedType, NumberFloatType, AllocatorType, JSONSerializer>::parse(IteratorType, IteratorType, nlohmann::basic_json<ObjectType, ArrayType, StringType, BooleanType, NumberIntegerType, NumberUnsignedType, NumberFloatType, AllocatorType, JSONSerializer>::parser_callback_t, bool) [with IteratorType = const char*; typename std::enable_if<std::is_base_of<std::random_access_iterator_tag, typename std::iterator_traits<_InputIterator>::iterator_category>::value, int>::type <anonymous> = 0; ObjectType = std::map; ArrayType = std::vector; StringType = std::__cxx11::basic_string<char>; BooleanType = bool; NumberIntegerType = long int; NumberUnsignedType = long unsigned int; NumberFloatType = double; AllocatorType = std::allocator; JSONSerializer = nlohmann::adl_serializer; nlohmann::basic_json<ObjectType, ArrayType, StringType, BooleanType, NumberIntegerType, NumberUnsignedType, NumberFloatType, AllocatorType, JSONSerializer>::parser_callback_t = std::function<bool(int, nlohmann::detail::parser<nlohmann::basic_json<> >::parse_event_t, nlohmann::basic_json<>&)>]’
/home/tle/src/pioneer/contrib/json/json.hpp:18867:33:   required from here
/home/tle/src/pioneer/contrib/json/json.hpp:3391:24: warning: comparison is always true due to limited range of data type [-Wtype-limits]
 3391 |             if ('\x00' <= c and c <= '\x1F')
      |                 ~~~~~~~^~~~
In file included from /usr/include/c++/11/bits/stl_algo.h:66,
                 from /usr/include/c++/11/functional:65,
                 from /usr/include/sigc++-2.0/sigc++/reference_wrapper.h:22,
                 from /usr/include/sigc++-2.0/sigc++/adaptors/bound_argument.h:24,
                 from /usr/include/sigc++-2.0/sigc++/adaptors/bind.h:6,
                 from /usr/include/sigc++-2.0/sigc++/adaptors/adaptors.h:23,
                 from /usr/include/sigc++-2.0/sigc++/sigc++.h:126,
                 from /home/tle/src/pioneer/src/libs.h:9,
                 from /home/tle/src/pioneer/src/graphics/VertexArray.h:8,
                 from /home/tle/src/pioneer/src/graphics/Drawables.h:8,
                 from /home/tle/src/pioneer/src/Background.h:7,
                 from /home/tle/src/pioneer/src/Intro.h:8,
                 from /home/tle/src/pioneer/src/Intro.cpp:4:
/usr/include/c++/11/bits/uniform_int_dist.h: In instantiation of ‘std::uniform_int_distribution<_IntType>::result_type std::uniform_int_distribution<_IntType>::operator()(_UniformRandomBitGenerator&, const std::uniform_int_distribution<_IntType>::param_type&) [with _UniformRandomBitGenerator = PiRngWrapper; _IntType = long unsigned int; std::uniform_int_distribution<_IntType>::result_type = long unsigned int]’:
/usr/include/c++/11/bits/stl_algo.h:3791:35:   required from ‘void std::shuffle(_RAIter, _RAIter, _UGenerator&&) [with _RAIter = __gnu_cxx::__normal_iterator<SceneGraph::Model**, std::vector<SceneGraph::Model*> >; _UGenerator = PiRngWrapper]’
/home/tle/src/pioneer/src/Intro.cpp:73:14:   required from here
/usr/include/c++/11/bits/uniform_int_dist.h:284:71: error: call to non-‘constexpr’ function ‘static unsigned int PiRngWrapper::min()’
  284 |         constexpr __uctype __urngmin = _UniformRandomBitGenerator::min();
      |                                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
/home/tle/src/pioneer/src/Intro.cpp:21:29: note: ‘static unsigned int PiRngWrapper::min()’ declared here
   21 |         static unsigned int min() { return 0; }
      |                             ^~~
In file included from /usr/include/c++/11/bits/stl_algo.h:66,
                 from /usr/include/c++/11/functional:65,
                 from /usr/include/sigc++-2.0/sigc++/reference_wrapper.h:22,
                 from /usr/include/sigc++-2.0/sigc++/adaptors/bound_argument.h:24,
                 from /usr/include/sigc++-2.0/sigc++/adaptors/bind.h:6,
                 from /usr/include/sigc++-2.0/sigc++/adaptors/adaptors.h:23,
                 from /usr/include/sigc++-2.0/sigc++/sigc++.h:126,
                 from /home/tle/src/pioneer/src/libs.h:9,
                 from /home/tle/src/pioneer/src/graphics/VertexArray.h:8,
                 from /home/tle/src/pioneer/src/graphics/Drawables.h:8,
                 from /home/tle/src/pioneer/src/Background.h:7,
                 from /home/tle/src/pioneer/src/Intro.h:8,
                 from /home/tle/src/pioneer/src/Intro.cpp:4:
/usr/include/c++/11/bits/uniform_int_dist.h:285:71: error: call to non-‘constexpr’ function ‘static unsigned int PiRngWrapper::max()’
  285 |         constexpr __uctype __urngmax = _UniformRandomBitGenerator::max();
      |                                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
/home/tle/src/pioneer/src/Intro.cpp:22:29: note: ‘static unsigned int PiRngWrapper::max()’ declared here
   22 |         static unsigned int max() { return std::numeric_limits<uint32_t>::max(); }
      |                             ^~~
In file included from /usr/include/c++/11/bits/stl_algo.h:66,
                 from /usr/include/c++/11/functional:65,
                 from /usr/include/sigc++-2.0/sigc++/reference_wrapper.h:22,
                 from /usr/include/sigc++-2.0/sigc++/adaptors/bound_argument.h:24,
                 from /usr/include/sigc++-2.0/sigc++/adaptors/bind.h:6,
                 from /usr/include/sigc++-2.0/sigc++/adaptors/adaptors.h:23,
                 from /usr/include/sigc++-2.0/sigc++/sigc++.h:126,
                 from /home/tle/src/pioneer/src/libs.h:9,
                 from /home/tle/src/pioneer/src/graphics/VertexArray.h:8,
                 from /home/tle/src/pioneer/src/graphics/Drawables.h:8,
                 from /home/tle/src/pioneer/src/Background.h:7,
                 from /home/tle/src/pioneer/src/Intro.h:8,
                 from /home/tle/src/pioneer/src/Intro.cpp:4:
/usr/include/c++/11/bits/uniform_int_dist.h:286:34: error: non-constant condition for static assertion
  286 |         static_assert( __urngmin < __urngmax,
      |                        ~~~~~~~~~~^~~~~~~~~~~
In file included from /home/tle/src/pioneer/src/Json.h:7,
                 from /home/tle/src/pioneer/src/JsonUtils.h:9,
                 from /home/tle/src/pioneer/src/ship/Propulsion.h:8,
                 from /home/tle/src/pioneer/src/ShipType.h:7,
                 from /home/tle/src/pioneer/src/Intro.h:10,
                 from /home/tle/src/pioneer/src/Intro.cpp:4:
/home/tle/src/pioneer/contrib/json/json.hpp: In instantiation of ‘std::string nlohmann::detail::lexer<BasicJsonType>::get_token_string() const [with BasicJsonType = nlohmann::basic_json<>; std::string = std::__cxx11::basic_string<char>]’:
/home/tle/src/pioneer/contrib/json/json.hpp:4564:57:   required from ‘void nlohmann::detail::parser<BasicJsonType>::parse(bool, BasicJsonType&) [with BasicJsonType = nlohmann::basic_json<>]’
/home/tle/src/pioneer/contrib/json/json.hpp:17252:79:   required from ‘static nlohmann::basic_json<ObjectType, ArrayType, StringType, BooleanType, NumberIntegerType, NumberUnsignedType, NumberFloatType, AllocatorType, JSONSerializer> nlohmann::basic_json<ObjectType, ArrayType, StringType, BooleanType, NumberIntegerType, NumberUnsignedType, NumberFloatType, AllocatorType, JSONSerializer>::parse(IteratorType, IteratorType, nlohmann::basic_json<ObjectType, ArrayType, StringType, BooleanType, NumberIntegerType, NumberUnsignedType, NumberFloatType, AllocatorType, JSONSerializer>::parser_callback_t, bool) [with IteratorType = const char*; typename std::enable_if<std::is_base_of<std::random_access_iterator_tag, typename std::iterator_traits<_InputIterator>::iterator_category>::value, int>::type <anonymous> = 0; ObjectType = std::map; ArrayType = std::vector; StringType = std::__cxx11::basic_string<char>; BooleanType = bool; NumberIntegerType = long int; NumberUnsignedType = long unsigned int; NumberFloatType = double; AllocatorType = std::allocator; JSONSerializer = nlohmann::adl_serializer; nlohmann::basic_json<ObjectType, ArrayType, StringType, BooleanType, NumberIntegerType, NumberUnsignedType, NumberFloatType, AllocatorType, JSONSerializer>::parser_callback_t = std::function<bool(int, nlohmann::detail::parser<nlohmann::basic_json<> >::parse_event_t, nlohmann::basic_json<>&)>]’
/home/tle/src/pioneer/contrib/json/json.hpp:18867:33:   required from here
/home/tle/src/pioneer/contrib/json/json.hpp:3391:24: warning: comparison is always true due to limited range of data type [-Wtype-limits]
 3391 |             if ('\x00' <= c and c <= '\x1F')
      |                 ~~~~~~~^~~~
make[2]: *** [CMakeFiles/pioneer-lib.dir/build.make:524: CMakeFiles/pioneer-lib.dir/src/Intro.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:473: CMakeFiles/pioneer-lib.dir/all] Error 2
make: *** [Makefile:149: all] Error 2
```

### Expected behaviour

No error

### Steps to reproduce

```
mkdir build && cd build
cmake ..
make -j32
```
__My pioneer version (and OS):__
Pioneer version: git SHA 76643bde166b0dc20bf386147b5089eda5e60f50
OS: Fedora 34

--- a/src/Intro.cpp
+++ b/src/Intro.cpp
@@ -18,8 +18,8 @@ public:
 	PiRngWrapper(size_t maxValue) :
 		maxVal(maxValue) {}
 	typedef unsigned int result_type;
-	static unsigned int min() { return 0; }
-	static unsigned int max() { return std::numeric_limits<uint32_t>::max(); }
+	static constexpr unsigned int min() { return 0; }
+	static constexpr unsigned int max() { return std::numeric_limits<uint32_t>::max(); }
 	unsigned int operator()()
 	{
 		return Pi::rng.Int32(maxVal);

